### ISSUE TITLE
Webpack doesn't rebuild UI when files in i18n directory change

### ISSUE THREAD

**Issue by yfain** [jhipster/generator-jhipster#6308]
##### **Overview of the issue**

If you generate the app without using internationalization, all the text of the UI is located in html files (components templates). Running the app in dev mode with `yarn start` works fine - webpack rebuilds the app and you can see the changes. 

If you use internationalization, the text of the UI is stored in the files in the directory in json files, e.g. webapp/i18n/en/. This directory is not watched by Webpack and you have to restart the dev server on any change. 

Some third-party Webpack plugin should be added to the webpack dev configuration to watch files in the webapp/i18n if the language support was selected during the app generation.

##### **Reproduce the error**

Generate an app with language support, start it in dev mode and try to change any text rendered in the home page.

##### **JHipster info**
Using JHipster version installed locally in current project's node_modules
Executing jhipster:info
Options: 
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
nyc-1@0.0.0 /Users/yfain11/Downloads/nyc1
└── generator-jhipster@4.7.0 

```
##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**
<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "gitHubOrganization": "",
    "baseName": "nyc1",
    "packageName": "io.github.jhipster.application",
    "serverPort": 8080,
    "serviceDiscoveryType": false,
    "authenticationType": "jwt",
    "uaaBaseName": "../uaa",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "useSass": false,
    "clientPackageManager": "yarn",
    "testFrameworks": [],
    "enableTranslation": false,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "ru"
    ],
    "clientFramework": "angularX",
    "jhipsterVersion": "4.7.0",
    "packageFolder": "io/github/jhipster/application",
    "jwtSecretKey": "replaced-by-jhipster-info",
    "jhiPrefix": "jhi"
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

git version 2.10.1

node: v8.1.3

npm: 5.3.0

yeoman: 2.0.0

yarn: 0.27.5

Docker version 17.03.1-ce, build c6d412e

docker-compose version 1.11.2, build dfed245

Execution complete

##### **Browsers and Operating System**

MAC OS 10.11.6,  Chrome 60

- [ x] Checking this box is mandatory (this is just to show you read everything)

**Comment by jsone-studios** [comment]
I had the same problem. I fixed this by updating the merge-jsons-webpack-plugin to the new version 1.0.11. The .json file change detection has been fixed in the latest updates: tettusud/merge-jsons-webpack-plugin#21

Now webpack recompiled on .json file changes. But the page was not reloaded automatically. I fixed this by adding `--watch-content-base` to the start script of the webpack-dev-server.
